lib/graphql/client.ts

```diff
@@ -1,14 +1,51 @@
-import { ApolloClient, HttpLink, InMemoryCache } from "@apollo/client";
+import { ApolloClient, createHttpLink, InMemoryCache, from } from "@apollo/client";
 import { registerApolloClient } from "@apollo/experimental-nextjs-app-support/rsc";
+import { onError } from '@apollo/client/link/error';
+
+const httpLink = createHttpLink({
+    uri: "http://localhost:3000/"
+});
+
+interface CustomError extends Error {
+    customInfo?: Record<string, any>;
+}
+
+const errorLink = onError(({ graphQLErrors, networkError, operation, forward }) => {
+    if (graphQLErrors) {
+        graphQLErrors.forEach(({ message, extensions, locations, path }) => {
+            //Check for specific error codes
+            switch (extensions?.code) {
+                case 'UNAUTHORIZED':
+                    console.error('Unauthorized');
+                    break;
+                case 'FORBIDDEN':
+                    console.error('Forbidden');
+                    break;
+                case 'GRAPHQL_VALIDATION_FAILED':
+                    extensions.customInfo = { customMessage: "There was a problem. Please try again" };
+                    operation.setContext({ extensions })
+                    break;
+                case 'INTERNAL_SERVER_ERROR':
+
+                    break;
+                default:
+                    console.error(`[GraphQL Error]: ${message}`)
+                    break;
+            }
+        })
+    }
+
+    if (networkError) {
+        const customNetworkError = networkError as CustomError;
+
+        customNetworkError.customInfo = { customMessage: 'There was a problem ' };
```

Comment:
This is just displaying the possibility of manipulating data or handling a specific error code differently using `errorLink`